src/js/player.js

```diff
@@ -2062,6 +2070,26 @@ class Player extends Component {
   }
 
   /**
+   * Returns the current source objects.
+   *
+   * @return {Object[]} The current source objects
+   * @method currentSources
+   */
+  currentSources() {
+    return this.cache_.sources || [this.currentSource()];
+  }
+
+  /**
+   * Returns the current source object.
+   *
+   * @return {Object} The current source object
+   * @method currentSource
+   */
+  currentSource() {
+    return this.cache_.source || { src: this.currentSrc() };
```

Comment:
I lean toward all three methods returning `undefined` if there is no source.

My second preference would be `{}` or `[]` for the two new ones, but I feel like in that case, it would make more sense for `currentSrc` to return `''` instead of `undefined`.